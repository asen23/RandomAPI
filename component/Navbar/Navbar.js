import Link from "next/link"
import styles from "./Navbar.module.css"

export default function Navbar() {
    return (
        <nav className={`${styles.container}`}>
            <Link href="/">
                <a className={`${styles.home}`}>
                    <h1>Random API</h1>
                </a>
            </Link>
            <div className={`${styles.linkGroup}`}>
                <Link href="/joke">
                    <a>
                        <div className={`${styles.link}`}>Joke</div>
                    </a>
                </Link>
                <Link href="/quote">
                    <a>
                        <div className={`${styles.link}`}>Quote</div>
                    </a>
                </Link>
            </div>
        </nav>
    )
}
