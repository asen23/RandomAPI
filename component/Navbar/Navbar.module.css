.container {
    background-color: ghostwhite;
    display: flex;
    justify-content: space-between;
    align-items: stretch;
    position: absolute;
    top: 0;
    width: 100%;
}

.linkGroup {
    display: flex;
    flex-direction: row-reverse;
    align-items: stretch;
    width: 80%;
}

.home {
    padding: 0 2%;
    text-align: center;
}

.linkGroup a {
    width: 8%;
}

.link {
    height: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
}

.link:hover,
.home:hover {
    background-color: antiquewhite;
}
