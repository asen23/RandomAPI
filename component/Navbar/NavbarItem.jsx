import Link from "next/link"

export default function NavbarItem(props) {
    return (
        <Link href={props.href}>
            <a className="px-4 flex justify-center items-center hover:bg-[antiquewhite]">{props.children}</a>
        </Link>
    )
}
