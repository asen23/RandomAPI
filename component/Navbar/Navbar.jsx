import Link from "next/link"
import NavbarItem from "./NavbarItem"

export default function Navbar() {
    return (
        <nav className="flex">
            <Link href="/">
                <a className="text-2xl p-4 font-bold">
                    <h1>Random API</h1>
                </a>
            </Link>
            <div className="flex flex-1 flex-row-reverse px-2">
                <NavbarItem href="/joke">Joke</NavbarItem>
                <NavbarItem href="/quote">Quote</NavbarItem>
                <NavbarItem href="/waifu">Waifu</NavbarItem>
                <NavbarItem href="/numberfact">Number Fact</NavbarItem>
            </div>
        </nav>
    )
}
