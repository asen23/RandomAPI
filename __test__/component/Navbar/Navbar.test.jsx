import { render, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event"
import Navbar from "component/Navbar/Navbar"

describe("Navbar", () => {
    it("renders unchanged", async () => {
        const { asFragment } = render(<Navbar />)
        expect(asFragment()).toMatchSnapshot()
    })
})
