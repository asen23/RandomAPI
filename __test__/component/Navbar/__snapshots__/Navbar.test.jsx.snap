// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders unchanged 1`] = `
<DocumentFragment>
  <nav
    class="flex absolute top-0 w-full bg-white"
  >
    <a
      class="text-2xl p-4 font-bold hover:bg-primary"
      href="/"
    >
      <h1>
        Random API
      </h1>
    </a>
    <div
      class="flex flex-1 flex-row-reverse pl-2"
    >
      <a
        class="px-4 flex justify-center items-center hover:bg-primary"
        href="/joke"
      >
        Joke
      </a>
      <a
        class="px-4 flex justify-center items-center hover:bg-primary"
        href="/quote"
      >
        Quote
      </a>
      <a
        class="px-4 flex justify-center items-center hover:bg-primary"
        href="/waifu"
      >
        Waifu
      </a>
      <a
        class="px-4 flex justify-center items-center hover:bg-primary"
        href="/numberfact"
      >
        Number Fact
      </a>
      <a
        class="px-4 flex justify-center items-center hover:bg-primary"
        href="/cat"
      >
        Cat
      </a>
    </div>
  </nav>
</DocumentFragment>
`;
